middle_end/flambda/inlining/inlining_decision.ml

```diff
@@ -49,14 +49,19 @@ module Function_declaration_decision = struct
         large_function_size: Code_size.t;
       }
 
-  let can_inline t =
+  type result =
+    | Not_inlinable
+    | Always_inlinable
```

Comment:
This case should rather be something along the lines of `Must_inline`, rather than simply "always inlinable".

Reply:
I've reworded everything to "Must_be_inlined", "Could_possibly_be_inlined" and "Can_not_be_inlined"